### DISCUSSION TITLE
Pipeline  Parameters are not implemented?

### DISCUSSION THREAD

**Discussion by brunskutscher** [sharpliner/sharpliner#207] [Q&A] [answered]
i cant find support for parameters on pipeline level.
is there a trick (interface implementation) to get parameters on pipeline level.
in my scenario pipeline parameters are required on pipeline parametrization (as input fields in run via portal).

thx 
matz

**Comment by premun** [comment]
This is currently not implemented, I think I've missed it in the schema somehow... Apologies for that!

However this is super easy to implement as we already have the grammar for this in template parameters.

I logged an issue (#208) and I should be able to get you going soon. Unless you're interested in implementing this?

**Reply by brunskutscher** [reply]
yes ... give me a try.
i will try to do this according the contribution guideline.
maybe i will come back later  ... and need some help.

thx so far

btw: the most ugliest way to define a "make/build/ant/etc ..." i ever faced in my last 30 years of development is the azure pipeline yaml stuff. it's time for a change!

**Reply by premun** [reply]
Great! Let me know, I am here to help.

Once you make your changes, please try to also write a unit test so that you get a feel for how it works. If you follow the documentation for defining templates, there are many useful macros in there and you will have to move those from `TemplateDefinition` to `PipelineDefinition`, I think, so that they are available.

**Comment by premun** [comment, marked as answer]
This has been implemented in #210
